managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -141,7 +141,11 @@ public void safeRun() {
         ManagedLedgerImpl.TOTAL_SIZE_UPDATER.addAndGet(ml, dataLength);
         if (ml.hasActiveCursors()) {
             // Avoid caching entries if no cursor has been created
-            ml.entryCache.insert(new EntryImpl(ledger.getId(), entryId, data));
+            EntryImpl entry = EntryImpl.create(ledger.getId(), entryId, data);
+            // EntryCache.insert: duplicates entry by allocating new entry and data. so, recycle entry after calling
+            // insert
+            ml.entryCache.insert(entry);
+            entry.recycle();
```

Comment:
you mean we should just insert same entry object to cache without copying it?
```
        if (ml.hasActiveCursors()) {
            EntryImpl entry = EntryImpl.create(ledger.getId(), entryId, data);
            ml.entryCache.insert(entry); // It should not duplicate and insert the same entry
        } else {
          data.release();
        }
```
If yes then I don't see any issue but then do you know any reason behind it having in existing [code](https://github.com/yahoo/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java#L144)..??